Sources/TuistGraph/Models/Target.swift

```diff
@@ -18,12 +18,12 @@ public struct Target: Equatable, Hashable, Comparable, Codable {
     // MARK: - Attributes
 
     public var name: String
-    public var platform: Platform
+    public var destinations: [Destination]
     public var product: Product
     public var bundleId: String
     public var productName: String
-    public var deploymentTarget: DeploymentTarget?
-
+    public var minimumDeployments: [MinimumDeployment]?
```

Comment:
Aren't `minimumDeployments` and `destination` strictly related?
For example, does it make sense to specify a minimumDeployment for `iOS` if there isn't a destination whose platform is iOS?

If it doesn't make sense, we should model it better, something like (just very quick proposal, havent thought much about it):
```
public enum Destination: String, Codable, Equatable, CaseIterable {
    case iOS(minVersion: String?, devices: IOSDevice)
    case macOS(minVersion: String?, devices: MacOSDevice)
    case tvOS(minVersion: String?)
    case watchOS(minVersion: String?)
}

public struct IOSDevice: OptionSet, Codable, Hashable {
    public static let iphone = IOSDevice(rawValue: 1 << 0)
    public static let ipad = IOSDevice(rawValue: 1 << 1)
    public static let macWithiPadDesign = IOSDevice(rawValue: 1 << 2)
}

public struct MacOSDevice: OptionSet, Codable, Hashable {
    public static let mac = MacOSDevice(rawValue: 1 << 0)
    public static let macCatalyst = MacOSDevice(rawValue: 1 << 1)
}
```

Reply:
My initial thought on this modeling was to follow the modeling from Xcode itself.  

A given destination implies an SDK (and thus needs a minimum deployment version). Each SDK can then only have a single version that is supported.

I wonder if a change like you suggested will make the downstream changes easier to manage, could `Destination` then own some of the logic around linting and graph traversal which currently rely on `target.platform` and `target.deploymentTarget`. it seems it could be a win for simplicity if this reduces the complexity from those split types.  I'll find some time to play with this approach. thanks

Reply:
I would try to keep our models close to what the data represents, and as strictly typed as possible, so I would go for the enum even if (not sure if it's the case) it makes slightly harder to implement tuist logic later

Reply:
Sounds good! 🙏